### PR TITLE
Integrate blockstream with ethers

### PR DESCRIPTION
**Pull request by justinbarry**
See: #1088

### CODE REVIEW

File: packages/augur-state/src/db/BlockstreamListener.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  }
+
+  onNewBlock = async (block: Block) => {
+    if(block) {
+      await this.deps.blockAndLogStreamer.reconcileNewBlock(block);
+    }
+  };
+
+  onLogsAdded = async (blockHash: string, logs: ExtendedLog[]) => {
+    const l:Log[] = logs.map((log) => ({
```
Comment (nuevoalex) [inline packages/augur-state/src/db/BlockstreamListener.ts]: can we name this something better than `l`?

File: packages/augur-state/src/db/BlockstreamListener.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      const logs = await provider.getLogs({
+        ...filterOptions,
+        topics: _.compact(filterOptions.topics)
+      });
+
+      return logs.map((log) => ({
+        ...log,
+        logIndex: (log.logIndex || "0").toString(),
+        blockNumber: (log.blockNumber || "0").toString(),
+        blockHash: log.blockHash || "0",
```
Comment (nuevoalex) [inline packages/augur-state/src/db/BlockstreamListener.ts:43]: when do these fallback values actually occur?

**Review by nuevoalex — APPROVED** [review]
One question and suggestion otherwise LGTM

File: packages/augur-state/src/db/BlockstreamListener.ts (hunk 2 above)
Comment (justinbarry) [reply packages/augur-state/src/db/BlockstreamListener.ts:43]: It makes typescript happy.

**Comment by pgebheim** [comment]
@justinbarry -- can you get the conflict sorted and get this merged first thing after the meeting?

**Comment by justinbarry** [comment]
@pgebheim FYI, I dropped the demo code.

**Pull request merged**